File: python/src/kalourmade_arkesel/error_mapping.py
from .exceptions import ApiError, AuthenticationError, InsufficientBalanceError, ValidationError


def raise_for_status(status: int, raw_body: str) -> None:
    if status < 400:
        return

    message = f"Arkesel API error (status {status})"

    if status == 402:
        raise InsufficientBalanceError(message)
    if status == 403:
        raise AuthenticationError(message)
    if status == 422:
        raise ValidationError(message)
    raise ApiError(status, raw_body, message)

File: node/src/errorMapping.ts
import { ApiError, AuthenticationError, InsufficientBalanceError, ValidationError } from './errors.js';

export function throwIfError(status: number, rawBody: string): void {
  if (status < 400) {
    return;
  }

  const message = `Arkesel API error (status ${status})`;

  if (status === 402) throw new InsufficientBalanceError(message);
  if (status === 403) throw new AuthenticationError(message);
  if (status === 422) throw new ValidationError(message);
  throw new ApiError(status, rawBody, message);
}
